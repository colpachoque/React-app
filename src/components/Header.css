.header {
    background-color: #b27bdb;
    width: 100%;
    min-height: 50px;
    display: flex;
    flex-direction: column;
    color: white;
}

.search {
    margin: 0 auto;
}