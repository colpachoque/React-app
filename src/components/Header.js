import React from 'react';
import styles from './Header.css';
import Search from './common/Search';

class Header extends React.Component {
    state = {
      searchText: ''
    };

    onSearchChange = (v) => {
        this.setState({ searchText: v})
    };
    render () {
        console.log(styles);
        return (
            <header className={styles.header}>
                <Search label={'FIND YOUR MOVIE'} value={this.state.searchText} onChange={this.onSearchChange}/>
                
            </header>
        );
    }
}

export default Header;