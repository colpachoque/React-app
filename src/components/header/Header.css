.header {
    background-color: #b27bdb;
    width: 100%;
    min-height: 50px;
    display: flex;
    flex-direction: column;
    color: white;
}

.search {
    width: 40%;
    margin: 0 auto;
}

.searchBySection {
    display: flex;
}

.radioGroup {
    width: 80%;
}

.searchByLabel {
    position: relative;
    top: 10px;
}

.results {
    background-color: wheat;
    color: #b27bdb;
    display: flex;
    align-items: center;
    min-height: 50px;
}

.resultsContainer {
    width: 40%;
    margin: 0 auto;
    display: flex;
}

.resultsInfo {
    width: 60%;
}

.resultsSort {
    width: 40%;
}
