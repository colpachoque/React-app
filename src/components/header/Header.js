import React from "react";
import styles from "./Header.css";
import Search from "../common/Search";
import Copyright from "../common/Copyright";
import Results from "../common/Results";
import MovieDetails from "../movie-details/MovieDetails";
import { updateSearchBy, searchInit } from "../../actions/search.action";
import { connect } from "react-redux";
import {
  HashRouter as Router,
  Route,
  Link,
  Switch,
  withRouter
} from "react-router-dom";
class Header extends React.Component {
  state = {
    searchText: ""
  };

  onSearchChange = searchValue => {
    this.setState({ searchText: searchValue });
  };

  onSearch = () => {
    this.props.searchInit(this.state.searchText);
  };

  onSearchFilterChange = filterValue => {
    this.props.updateSearchBy(filterValue);
  };

  render() {
    return (
      <header className={styles.header}>
        {!this.props.location.pathname.includes("/movie/") && (
          <>
            <Copyright />
            <Search
              label={"FIND YOUR MOVIE"}
              value={this.state.searchText}
              searchBy={this.props.searchBy}
              onChange={this.onSearchChange}
              onSearch={this.onSearch}
              onSearchFilterChange={this.onSearchFilterChange}
            />
          </>
        )}
        <Route exact path="/movie/:id" component={MovieDetails} />
        <Results info={`${this.props.moviesFound} films found`} />
      </header>
    );
  }
}

const mapStateToProps = ({ search: { searchBy, moviesFound } }) => {
  return { searchBy, moviesFound };
};
export default withRouter(
  connect(
    mapStateToProps,
    { updateSearchBy, searchInit }
  )(Header)
);
