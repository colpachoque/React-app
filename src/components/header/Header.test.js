import React from "react";
import Header from "./Header";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
describe("Header component", () => {
  let movie;

  beforeAll(() => {
    movie = {
      name: "The best movie ever",
      genre: "Drama",
      year: 1994
    };
  });

  it("should be render correctly", () => {
    const component = shallow(<Header />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
