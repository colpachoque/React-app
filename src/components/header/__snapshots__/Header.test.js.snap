// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should be render correctly 1`] = `
<header
  className="header"
>
  <Copyright />
  <Search
    label="FIND YOUR MOVIE"
    onChange={[Function]}
    onSearch={[Function]}
    onSearchFilterChange={[Function]}
    searchBy="title"
    value=""
  />
  <Results
    info="10 films found"
  />
</header>
`;
